src/compiler/crystal/compiler.cr

```diff
@@ -576,12 +576,26 @@ module Crystal
     end
 
     protected def optimize(llvm_mod)
+      {% if LibLLVM::IS_LT_130 %}
+        optimize_with_old_pass_manager(llvm_mod)
+      {% else %}
+        optimize_with_new_pass_manager(llvm_mod)
+      {% end %}
+    end
+
+    private def optimize_with_old_pass_manager(llvm_mod)
       fun_pass_manager = llvm_mod.new_function_pass_manager
       pass_manager_builder.populate fun_pass_manager
       fun_pass_manager.run llvm_mod
       module_pass_manager.run llvm_mod
     end
 
+    private def optimize_with_new_pass_manager(llvm_mod)
+      LLVM::PassBuilderOptions.new do |options|
+        LLVM.run_passes(llvm_mod, "default<O3>", target_machine, options)
```

Comment:
We could make this "default<03>" string configurable, to experiment with different passes and options.